### DISCUSSION TITLE
Custom RETRY per Service

### DISCUSSION THREAD

**Discussion by carnei-ro** [Kong/kong#7338] [Ideas and feature requests]
So I have this app ([image here](docker.io/leandrocarneiro/request-logger:env-and-proxy)), I can do a GET request to path `/fail/<N>` where **N** is the percentage of chances to the app return 503 error.  
Using Istio 1.9 with default installation and no configuring VirtualService to explicitly implement retries, the envoy proxies do retries out-of-the-box for me.
e.g:
```
kubectl exec -it --context="${CTX_CLUSTER1}" -n sample -c sleep     "$(kubectl get pod --context="${CTX_CLUSTER1}" -n sample -l    app=sleep -o jsonpath='{.items[0].metadata.name}')"     -- sh -c "for i in \$(seq 1 100); do curl -sS request-logger.sample:8000/fail/30; echo ""; done" | sort | uniq -c
      4 forced fail
     96 ok
```
It means: Although the app had 30% chances to fail, I only got 4 of 100 responses with status 503.

Using Kong (just service and route) the "retries" (default to 5) seems to not considering the HTTP Status Code to perform the retry:
```
for i in $(seq 1 100); do curl -sS localhost:8000/; echo ""; done | sort | uniq -c
     31 forced fail
     69 ok
```
For the same 30% of chances to fail, I got 31 503 responses. (fit the 30% odds).

I've made it work setting the env var KONG_NGINX_PROXY_PROXY_NEXT_UPSTREAM="error timeout http_503", but the behavior will be applied to all requests. Is it possible to configure just to one service? Imagine some of my services I want to change to "http_500" and set "non_idempotent".

**Comment by obliadp** [comment]
I would love to see proxy-next-upstream options  be adjustable per route/service! Any development on this?
